content/en/articles/tdd/index.md

```diff
@@ -0,0 +1,294 @@
+---
+date: "2017-01-07"  
+title: "Getting started with TDD"  
+tags:
+  - tdd
+  - bdd
+  - ci  
+authors:
+  - Freezystem
+---
+
+## Introduction
+
+Everything begins in 1999, when [Kent Beck](https://en.wikipedia.org/wiki/Kent_Beck) 
+present a new software development methodology based on AGILE: the eXtreme Programmming aka _XP_.  
+_XP_ define new optimized development practices improving code production and strength.  
+Most of these methodology principles are yet well-known among which we can highlight 
+**Continuous Integration** and **pair programming**.
+
+The point that pick our interest here is another pillar of his methodology that consists of
+driving development with tests, you named it : _TDD_.
+
+TDD is a hot topic because it's costly to set up.
+Despite of its growing popularity it can barely find its way.
+
+With the multiplication of running environment, the complexity of web applications
+and the rising of Open-Source projects. Maintainers are facing cross compatibility issues
+and inconsistency.  
+Nowadays, standards are more and more permissive and favors risky behaviours. 
+Most of the time systems are able to correct themselves your errors, sometimes 
+without even warning you.
+Within this context:
+- How to flawlessly write functional cross-platform code ?
+- How to be sure that the newly added code isn't breaking the old one ?
+- How to be sure that the code behave like it's expected to ? 
+
+Starting by adopting strict coding practices is a must
+but tests will be crucial to your code quality.
+
+Fortunately TDD is bringing an elegant solution to these problems.
+
+## Did you just say TDD ?
+
+Test Driven Development, is a development technique that enforce tests writing
+before even writing a single line of code.
+
+In theory, this method requires the attendance of at least two different coders,
+one to write the tests and the other write the tested code. That allowing issues
+linked to subjectivity.
+
+In practice things are more complicated, sometimes we are developing alone or
+we are writing by ourselves the tests that ensure the integrity of a new feature
+in a collaborative project.
+
+> Whatever happens, a low-efficient test is more valuable that no test at all.
+> The aim being to be used to write some and to be objective in their writing.
```

Comment:
je vais mettre `The aim is to get used to write some test and to be objective about their writing.`